src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1611,10 +1705,13 @@ private SolutionState WithDocumentState(DocumentState newDocument, bool textChan
                 return this;
             }
 
-            return this.TouchDocument(newDocument.Id, p => p.UpdateDocument(newDocument, textChanged, recalculateDependentVersions));
+            return this.TouchDocument(newDocument.Id, filePathsToDocumentIds, p => p.UpdateDocument(newDocument, textChanged, recalculateDependentVersions));
         }
 
         private SolutionState TouchDocument(DocumentId documentId, Func<ProjectState, ProjectState> touchProject)
+        => TouchDocument(documentId, _filePathToDocumentIdsMap, touchProject);
+
+        private SolutionState TouchDocument(DocumentId documentId, ImmutableDictionary<string, ImmutableArray<DocumentId>> filePathsToDocumentIds, Func<ProjectState, ProjectState> touchProject)
```

Comment:
I would have thought TouchDocument only gets called in places where we're mutating a document and not renaming it?

Reply:
`WithDocumentState` calls this, which ends up calling `ProjectState.UpdateDocument` as the func.